src/main/java/stsjorbsmod/cards/wanderer/Amnesia.java

```diff
@@ -38,9 +41,19 @@ public Amnesia() {
 
     @Override
     public void use(AbstractPlayer p, AbstractMonster _) {
-        if(dontTriggerOnUseCard) {
-            addToBot(new SnapAction(p));
-            addToBot(new RemovePowersMatchingPredicateAction(p, power -> power.type == AbstractPower.PowerType.BUFF || power.type == AbstractPower.PowerType.DEBUFF));
+        if (dontTriggerOnUseCard) {
+            List<AbstractPower> buffs = p.powers.stream()
+                    .filter(power -> power.type == AbstractPower.PowerType.BUFF)
+                    .collect(Collectors.toList());
+            if (buffs.size() == 0) {
+                AbstractDungeon.actionManager.addToBottom(new TalkAction(true,"~...~", 2.0f, 2.0f));
+            } else if (p.hasPower(ArtifactPower.POWER_ID)) {
+                AbstractDungeon.actionManager.addToBottom(new ReducePowerAction(p, p, ArtifactPower.POWER_ID, 1));
+            } else {
+                int index = AbstractDungeon.miscRng.random(buffs.size());
+                AbstractPower powerToRemove = buffs.get(index);
+                AbstractDungeon.actionManager.addToBottom(new ReducePowerAction(p, p, powerToRemove.ID, powerToRemove.amount));
```

Comment:
@Bobertg#0304 Is there some specific reason to prefer `new ReducePowerAction(p, p, powerToRemove.ID, powerToRemove.amount)` vs `new RemoveSpecificPowerAction(p, p, powerToRemove.ID)`? The latter feels more readable if there's no intended mechanical difference

Reply:
I remember having issues with RemoveSpecificPowerAction in the past, which is why I didn't do it here. But maybe I was doing something wrong then.